src/modules/cron/cron.controller.ts

```diff
@@ -178,4 +183,49 @@ export class CronController {
             throw new Error('Failed to remove users due to an internal server error.');
         }
     }
+
+    @Put('unlock')
+    @ApiOkResponse({
+        description: 'The users were successfully unlocked.',
+        type: Boolean,
+    })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to unlock users.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to unlock users.' })
+    @ApiNotFoundResponse({ description: 'Insufficient permissions to unlock users.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while trying to unlock users.' })
+    public async unlockUsersWithExpiredLocks(@Permissions() permissions: PersonPermissions): Promise<boolean> {
+        try {
+            const userLocks: UserLock[] = await this.userLockRepository.getLocksToUnlock();
+            if (userLocks.length === 0) {
+                return true;
+            }
+
+            const results: PromiseSettledResult<Result<void, DomainError>>[] = await Promise.allSettled(
+                userLocks.map(async (userLock: UserLock) => {
+                    const person: Result<Person<true>> = await this.personRepository.getPersonIfAllowed(
+                        userLock.person,
+                        permissions,
+                    );
+                    if (!person.ok || !person.value.keycloakUserId) {
+                        return { ok: false, error: new EntityNotFoundError() };
+                    }
+                    return this.keyCloakUserService.updateKeycloakUserStatus(
+                        person.value.id,
+                        person.value.keycloakUserId,
+                        userLock,
+                        false,
+                    );
+                }),
+            );
+
+            const allSuccessful: boolean = results.every(
```

Comment:
I guess we use Promise.allSettled instead of Promise.all here because we still want to unlock another user even though the previous one/ones fail but apparently the endpoint only returns true if all promises resolve successfully so we might get an error even though we unlocked 1000 users and only 1 was errorful... Not sure if that's good and helpful but we might need to refactor all these endpoints if we want more informations and metrics in the future.